src/ResourceManager/Insights/Commands.Insights/Metrics/GetAzureRmMetricCommand.cs

```diff
@@ -39,33 +42,42 @@ public class GetAzureRmMetricCommand : InsightsClientCmdletBase
         /// <summary>
         /// Gets or sets the ResourceId parameter of the cmdlet
         /// </summary>
-        [Parameter(Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource Id")]
+        [Parameter(ParameterSetName = GetAzureRmAMetricParamGroup, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource Id")]
+        [Parameter(ParameterSetName = GetAzureRmAMetricFullParamGroup, Position = 0, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource Id")]
         [ValidateNotNullOrEmpty]
         public string ResourceId { get; set; }
 
         /// <summary>
         /// Gets or sets the timegrain parameter of the cmdlet
         /// </summary>
-        [Parameter(Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The time grain of the query")]
+        [Parameter(ParameterSetName = GetAzureRmAMetricFullParamGroup, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The time grain of the query.")]
```

Comment:
This was one of the real breaking changes. 
In this API if MetricNames is present, then all the other parameters can be present.
If there is not MetricNames no other filter is valid, except for resourceId.

It did not make too much sense (IMO) to leave timeGrain as Pos = 1 when it can only be used when MetricNames is used. BTW: timeGrain is optional now and MetricNames is Pos = 1.